# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: c2741585119b
# Full Hash: c2741585119ba8536918b94429296479c0865c25
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-07-27 09:45:43
# Regressor Bug: 1695806
# File Overlap Count: 1
# Description:
#   Bug 1695806 - Defer sending a snapshot reqeust for remote process iframe until cloning the iframe document finished. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119950
# ==============================================================================

diff -r 3c5186df3677 -r c2741585119b dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Tue Jul 27 00:47:44 2021 +0000
+++ b/dom/ipc/ContentParent.cpp	Tue Jul 27 00:47:44 2021 +0000
@@ -3844,20 +3844,8 @@
     return IPC_OK();
   }
 
-  RemotenessChangeOptions options;
-  options.mRemoteType = cp->GetRemoteType();
-  target->ChangeRemoteness(options, /* aPendingSwitchId = */ 0)
-      ->Then(
-          GetMainThreadSerialEventTarget(), __func__,
-          [source = RefPtr{source},
-           data = std::move(aPrintData)](BrowserParent* aBp) {
-            Unused << aBp->SendCloneDocumentTreeIntoSelf(source, data);
-          },
-          [](nsresult aRv) {
-            NS_WARNING(
-                nsPrintfCString("Remote clone failed: %x\n", unsigned(aRv))
-                    .get());
-          });
+  target->CloneDocumentTreeInto(source, cp->GetRemoteType(),
+                                std::move(aPrintData));
   return IPC_OK();
 }
 